RHDEVDOCS-6010: Content creation for GitOps 1.13.0 RN (#77471)
openshift/openshift-docs: modules/gitops-release-notes-1-13-0.adoc

* link:https://access.redhat.com/errata/RHSA-XXXX:NNNN[RHSA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="new-features-1-13-0_{context}"]
== New features

* With this update, you can configure Argo CD to disable the automatic scraping of metrics for your instance. Before this update, there was no option to disable this feature, resulting in excessive storage usage when multiple Argo CD instances were available on the cluster. link:https://issues.redhat.com/browse/GITOPS-4519[GITOPS-4519]
+
[NOTE]
====
In {gitops-shortname}, the scraping of metrics is enabled by default for all Argo CD instances.
====
+
To disable metrics for your Argo CD instance, edit the `spec.monitoring.disableMetrics` field in the Argo CD custom resource to `false`.
+
.Example 
[source,YAML]
----
apiVersion: argoproj.io/v1beta1
kind: Argo CD
metadata:
 name: Argo CD-sample
 namespace: default
spec:
 monitoring:
   disableMetrics: true
----
+

> Use the following command to scrape metrics in the Argo CD CLI.

Author: Fixed.

+

> .Example

Author: Fixed.

[source,terminal]
----
oc patch Argo CD Argo CD-sample -n default --type='json' -p='[{"op": "replace", "path": "/spec/monitoring/disableMetrics", "value": true}]'
----

> * With this update, the Argo Rollouts functionality in {gitops-title} is no longer supported as a Technology Preview (TP) feature and it is promoted to the General Availability (GA). Use this feature to:

Author: Fixed.

** Run cluster-scoped Argo Rollouts instead of namespace-scoped
** Route traffic using Openshift Routes for canary application deployment 

> ** Configure Argo Rollouts CLI 

Author: Fixed.

> For more information, see "Argo Rollouts". link:https://issues.redhat.com/browse/GITOPS-3848[GITOPS-3848]

Author: Fixed.

> * With this update, users can perform canary-based Argo Rollouts deployments using OpenShift Routes-based traffic management. Argo Rollouts supports the ability to split traffic using Routes, by modifying the percentage of traffic of the Route that reaches the canary or stable versions of the application, during a deployment. link:https://issues.redhat.com/browse/GITOPS-2400[GITOPS-2400]

Author: Fixed.

> * With this update, the Argo CD Applications in any namespace feature is no longer supported as a technology preview (TP) feature and it is promoted to the General Availability (GA). link:https://issues.redhat.com/browse/GITOPS-3796[GITOPS-3796]

Author: Fixed.

> * With this update, users can customize permissions for a cluster-scoped Argo CD instance by creating user defined `ClusterRoles` and `ClusterRoleBindings` for the Argo CD application controller and Server components. link:https://issues.redhat.com/browse/GITOPS-2614[GITOPS-2614]

Author: Fixed.

[id="fixed-issues-1-13-0_{context}"]
== Fixed issues

> * Before this update, if the `DeploymentConfig` API, `DeploymentConfig [apps.openshift.io/v1]`, was disabled in the {OCP} cluster, the {gitops-shortname} Operator enters the `CrashLoopBackoff` error state and does not function properly. This update fixes the issue by modifying the functionality of the {gitops-shortname} Operator to ensure that it does not enter the `CrashLoopBackoff` error state when the API is unavailable in the cluster. link:https://issues.redhat.com/browse/GITOPS-4489[GITOPS-4489]

Author: Fixed.

* Before this update, users were unable to connect to remote Git repositories through a SOCKS5 proxy server because it was only compatible with HTTPS URLs. With this update, users can access remote Git repositories by using a SOCKS5 proxy server that supports the SSH protocol. link:https://issues.redhat.com/browse/GITOPS-3710[GITOPS-3710]

> * Before this update, the `forceHttpBasicAuth` repository configuration was not initialized properly during repository updates from Argo CD Web UI. As a result, the value was reset to default, leading to connection errors, particularly for Azure DevOps repositories. With this update, a change has been made in the upstream to ensure that the `forceHttpBasicAuth` configuration is correctly initialized during repository update operations. link:https://issues.redhat.com/browse/GITOPS-3706[GITOPS-3706]

Author: Fixed.

> * Before this update, resource deletion was not handled properly in the Argo CD upstream. With this update, Argo CD upstream has been enhanced to ensure that resource deletion is managed properly. Now, Argo CD will wait for resources to be entirely deleted before advancing to the next sync-wave in the deletion process. This approach ensures a controlled and predictable workflow, enhancing the user experience and preventing potential issues that may arise from not waiting for resources to be fully recycled before proceeding to the next sync-wave. link:https://issues.redhat.com/browse/GITOPS-2642[GITOPS-2642]

Author: Fixed.

* Before this update, the Argo CD CLI would designate the initial source as the main source for multi-source applications. With this update, the Argo CD CLI supports multi-source applications by considering all sources during processing. link:https://issues.redhat.com/browse/GITOPS-2623[GITOPS-2623]

> * Before this update, {gitops-title} Operator used an older version of the HA proxy router container image. This container image had multiple security vulnerabilities. With this update, the {gitops-title} Operator uses the latest version of the HA proxy container image, which includes a fix for the security vulnerabilities. link:https://issues.redhat.com/browse/GITOPS-4761[GITOPS-4761]

Author: The SME has instructed us to remove this issue from this release and move it to the v1.12.4 z-stream release. Hence, I am removing this entry from this content.

[id="deprecated-features-1-13-0_{context}"]
== Deprecated and removed features

> * In {gitops-title} 1.13, {gitops-title} Application Manager command-line interface (CLI), `kam` is deprecated and is planned to be removed in the {gitops-title} 1.14. Red Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to {gitops-title} Application Manager command-line interface (CLI), `kam`, you can use <alternative_capability_or_feature_if_available> instead.

Reviewer: Note: the {nbsp} thing is non-breaking space that should be always added between Red and Hat :+1:  See: https://redhat-documentation.github.io/supplementary-style-guide/#non-breaking-spaces

> * In {gitops-title} 1.13, {gitops-title} Application Manager command-line interface (CLI), `kam` is deprecated and is planned to be removed in the {gitops-title} 1.14. Red Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to {gitops-title} Application Manager command-line interface (CLI), `kam`, you can use <alternative_capability_or_feature_if_available> instead.

Author: Thanks for educating me about the advisories in the first two options. I have gone ahead and filled them in.

* Before this update, users were unable to connect to remote Git repositories through a SOCKS5 proxy server because it was only compatible with HTTPS URLs. With this update, users can access remote Git repositories by using a SOCKS5 proxy server that supports the SSH protocol. link:https://issues.redhat.com/browse/GITOPS-3710[GITOPS-3710]

* Before this update, the `forceHttpBasicAuth` repository configuration was not initialized properly during repository updates from Argo CD Web UI. As a result, the value was reset to default, leading to connection errors, particularly for Azure DevOps repositories. This update fixes the issue by adding a change in the upstream to ensure that the `forceHttpBasicAuth` configuration is correctly initialized during repository update operations. link:https://issues.redhat.com/browse/GITOPS-3706[GITOPS-3706]

> * Before this update, resource deletion was not handled properly in the Argo CD upstream. With this update, Argo CD upstream is enhanced to ensure that resource deletion is managed properly. Now, Argo CD waits for resources to be entirely deleted before advancing to the next sync-wave in the deletion process. This approach ensures a controlled and predictable workflow, enhancing the user experience and preventing potential issues that might arise from not waiting for resources to be fully recycled before proceeding to the next sync-wave. link:https://issues.redhat.com/browse/GITOPS-2642[GITOPS-2642]

Reviewer 2: Can we move this to new features section? and a small update to text.
```suggestion
* With this update, Argo CD upstream is enhanced to ensure that resource deletion is managed properly when using sync-waves. Now, Argo CD waits for resources to be entirely deleted before advancing to the next sync-wave in the deletion process. This approach ensures a controlled and predictable workflow, enhancing the user experience and preventing potential issues that might arise from not waiting for resources to be fully recycled before proceeding to the next sync-wave. link:https://issues.redhat.com/browse/GITOPS-2642[GITOPS-2642]
```

Author: Yes, I have now included this RN text in the Features section.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-13-0_{context}"]
= Release notes for {gitops-title} 1.13.0

{gitops-title} 1.13.0 is now available on {OCP} 4.12, 4.13, 4.14, 4.15, and 4.16.

[id="errata-updates-1-13.0_{context}"]
== Errata updates

[id="RHSA-XXXX:NNNN-gitops-1-13-0-security-update-advisory_{context}"]
=== RHSA-XXXX:NNNN - {gitops-title} 1.13.0 security update advisory

Issued:

The list of security fixes that are included in this release is documented in the following advisory:

* link:https://access.redhat.com/errata/RHSA-XXXX:NNNN[RHSA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="new-features-1-13-0_{context}"]
== New features

> * With this update, you can configure Argo CD to disable the automatic scraping of metrics for your instance. Before this update, there was no option to disable this feature, resulting in excessive storage usage when multiple Argo CD instances were available on the cluster. link:https://issues.redhat.com/browse/GITOPS-4519[GITOPS-4519]

Reviewer 3: Just wanted to add this comment to verify if we use Argo CD or ArgoCD - whatever is used for consistency in the docs. 
Otherwise LGTM, thanks @Dhruv-Soni11

Author: Hi @saumeya - I just quickly searched for instances of Argo CD and ArgoCD in the previous release notes and I found out that there were more occurrences of the term Argo CD used in the GitOps documentation in paragraphs and instances of ArgoCD are mostly likely found in the code snippet. So, I guess it makes sense to stick to Argo CD for release notes text and ArgoCD in code blocks or snippets. WDYT?

Reviewer 3: Awesome, yes makes sense then.
There is one change I had requested on line 41, can you revert that back to Argo CD.

Reviewer 2: The way I differentiate between these is, if I am talking about something in ArgoCD CRD/CR maybe a field or some config, I use term `ArgoCD` and if its something regarding Argo CD maybe a feature or configurations, I use `Argo CD`.

+
[NOTE]
====
In {gitops-shortname}, the scraping of metrics is enabled by default for all Argo CD instances.
====
+

> To disable metrics for your Argo CD instance, edit the `spec.monitoring.disableMetrics` field in the Argo CD custom resource to `false`.

Reviewer 3: Update the last part to - 

field in the ArgoCD CR to `true`
instead of
Argo CD custom resource to `false`

Author: Fixed.

+
.Example 
[source,YAML]
----
apiVersion: argoproj.io/v1beta1

> kind: Argo CD

Reviewer 3: kind: ArgoCD (no space between argo and cd)

Author: Fixed.

metadata:

>  name: Argo CD-sample

Reviewer 3: name: argocd-sample (no spaces and small makes it more readable)

Author: Fixed.

 namespace: default
spec:
 monitoring:
   disableMetrics: true
----
+
Use the following command to scrape metrics in the Argo CD CLI:
+
[source,terminal]
----

> $ oc patch Argo CD Argo CD-sample -n default --type='json' -p='[{"op": "replace", "path": "/spec/monitoring/disableMetrics", "value": true}]'

Reviewer 3: oc patch argocd <argocd-instance-name> -n <namespace> -- type='json' .....

if we don't want generics in documentation then
oc patch argocd argocd-sample -n ....

Author: Fixed.

----

> * With this update, the Argo Rollouts functionality in {gitops-title} is promoted from Technology Preview (TP) to General Availability (GA). Use this feature to perform the following actions: link:https://issues.redhat.com/browse/GITOPS-3848[GITOPS-3848]

Author: Fixed.

> ** Run cluster-scoped Argo Rollouts instead of namespace-scoped

Author: Fixed.

> ** Route traffic by using Openshift Routes for canary application deployment 

Author: Fixed.

> ** Configure Argo Rollouts CLI

Author: Fixed.

* With this update, you can perform canary-based Argo Rollouts deployments by using OpenShift Routes-based traffic management. Argo Rollouts supports traffic splitting by modifying the percentage of traffic that reaches the canary or stable application version during a deployment. link:https://issues.redhat.com/browse/GITOPS-2400[GITOPS-2400]

* With this update, the Argo CD Applications in any namespace feature is promoted from Technology Preview (TP) to General Availability (GA). link:https://issues.redhat.com/browse/GITOPS-3796[GITOPS-3796]

* With this update, the following new commands are introduced for the Apps in any namespace feature:

> ** `argocd proj add-source-namespace` - Use this command to add source namespaces to `AppProject.Spec.SourceNamespaces`, for whitelisting specific namespaces for application creation.

Author: Fixed.

> ** `argocd proj delete-source-namespace` - Use this command to remove source namespaces from `AppProject.Spec.SourceNamespaces`, and for blacklisting specific namespaces. link:https://issues.redhat.com/browse/GITOPS-4045[GITOPS-4045]

Author: Fixed.

> ** `argocd proj delete-source-namespace` - Use this command to remove source namespaces from `AppProject.Spec.SourceNamespaces`, for blacklisting specific namespaces. link:https://issues.redhat.com/browse/GITOPS-4045[GITOPS-4045]

Reviewer 4: small nit- I think you're missing maybe an "and" before "for" after the comma? It doesn't read clearly to me

Author: Yes, the addition of the and does make the sentence more readable. Thanks for catching this nit. Fixed.

* With this update, you can customize permissions for a cluster-scoped Argo CD instance by creating user-defined `ClusterRoles` and `ClusterRoleBindings` for the Argo CD Application Controller and server components. link:https://issues.redhat.com/browse/GITOPS-2614[GITOPS-2614]

* With this update, you can customize the Keycloak route name in the {gitops-title} Operator. This update also enables support for complex routing scenarios, such as dividing routes among various ingress controller shards. link:https://issues.redhat.com/browse/GITOPS-4451[GITOPS-4451]

> * With this update, the default TLS termination policy of the Argo CD server's Route changed from `Passthrough` to `Reencrypt`. As a result, the TLS connection now relies on the default ingress operator certificate instead of the self-signed certificate generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]

Reviewer 5: ```suggestion
* With this update, the default TLS termination policy of the Argo CD server's Route changed from `Passthrough` to `Reencrypt`. As a result, the TLS connection relies on the default ingress Operator certificate instead of the self-signed certificate generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]
```
https://redhat-documentation.github.io/supplementary-style-guide/#operator

Author: Fixed.

> * With this update, the default TLS termination policy of the Argo CD instances changed from `Passthrough` to `Reencrypt`. As a result, the TLS connection now relies on the default ingress operator certificate instead of the self-signed certificates generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]

Reviewer 6: Nit: This sounds more accurate.
```suggestion
* With this update, the default TLS termination policy of the Argo CD server's Route changed from `Passthrough` to `Reencrypt`. As a result, the TLS connection now relies on the default ingress operator certificate instead of the self-signed certificate generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]
```

Author: Fixed.

* With this update, Argo CD upstream is enhanced to ensure that resource deletion is managed properly when using sync-waves. Now, Argo CD waits for resources to be entirely deleted before advancing to the next sync-wave in the deletion process. This approach ensures a controlled and predictable workflow, enhancing the user experience and preventing potential issues that might arise from not waiting for resources to be fully recycled before proceeding to the next sync-wave. link:https://issues.redhat.com/browse/GITOPS-2642[GITOPS-2642]

> * With this update, Argo CD CLI has been enhanced by enabling the `--app-namespace` flag for various application-related commands. This flag allows users to specify the namespace for operations such as waiting, rolling back, patching, editing, diffing, viewing history, syncing, and deleting applications. As a result, the following commands are updated in {gitops-title}: link:https://issues.redhat.com/browse/GITOPS-3834[GITOPS-3834]

Reviewer 5: Would suggest moving Jira ticket to after first sentence to avoid breaking up this list.

Author: Fixed.

** `argocd app wait`
** `argocd app rollback`
** `argocd app patch`
** `argocd app edit`
** `argocd app diff`
** `argocd app history`
** `argocd app sync`
** `argocd app delete`
** `argocd app get`

[id="fixed-issues-1-13-0_{context}"]
== Fixed issues

* Before this update, if the `DeploymentConfig [apps.openshift.io/v1]` API was disabled in the {OCP} cluster, the {gitops-shortname} Operator would enter the `CrashLoopBackoff` error state and would not function properly. This update fixes the issue by modifying the functionality of the {gitops-shortname} Operator to ensure that it does not enter the `CrashLoopBackoff` error state when the API is unavailable in the cluster. link:https://issues.redhat.com/browse/GITOPS-4489[GITOPS-4489]

* Before this update, users were unable to connect to remote Git repositories through a SOCKS5 proxy server because it was only compatible with HTTPS URLs. With this update, users can access remote Git repositories by using a SOCKS5 proxy server that supports the SSH protocol. link:https://issues.redhat.com/browse/GITOPS-3710[GITOPS-3710]

> * Before this update, the `forceHttpBasicAuth` repository configuration was not initialized properly during repository updates from Argo CD Web UI. As a result, the value was reset to default, leading to connection errors, particularly for Azure DevOps repositories. This update fixes the issue by adding a change in the upstream to ensure that the `forceHttpBasicAuth` configuration is correctly initialized during repository update operations. link:https://issues.redhat.com/browse/GITOPS-3706[GITOPS-3706]

Reviewer 5: ```suggestion
* Before this update, the `forceHttpBasicAuth` repository configuration was not initialized properly during repository updates from Argo CD UI. As a result, the value was reset to default, leading to connection errors, particularly for Azure DevOps repositories. This update fixes the issue by adding a change in the upstream to ensure that the `forceHttpBasicAuth` configuration is correctly initialized during repository update operations. link:https://issues.redhat.com/browse/GITOPS-3706[GITOPS-3706]
```

Author: Fixed.

> * With this update, the default TLS termination policy of the Argo CD instances changed from `Passthrough` to `Reencrypt`. This update fixes the issue because the TLS connection now relies on the default ingress operator certificate instead of the self-signed certificates generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]

Reviewer 6: Since this is an enhancement to an existing feature, can we move this to the Feature section?
```suggestion
* With this update, the default TLS termination policy of the Argo CD instances changed from `Passthrough` to `Reencrypt`. As a result, the TLS connection now relies on the default ingress operator certificate instead of the self-signed certificates generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]
```

Author: Yes, we can. I have fixed your comment @chetan-rns. PTAL.

Reviewer 6: Looks good. Thanks @Dhruv-Soni11

> * Before this update, the RolloutManager controller did not create ServiceMonitor. With this update, the RolloutManager controller is improved to include support for ServiceMonitor creation. As a result, you can now gather metrics from Rollouts using the enhanced RolloutManager controller, and these metrics are visible in monitoring. link:https://issues.redhat.com/browse/GITOPS-3271[GITOPS-3271]

Reviewer 5: Not sure if ServiceMonitor should be in backticks, if it is an object.

Author: Fixed.

> * Before this update, the RolloutManager controller did not create ServiceMonitor. With this update, the RolloutManager controller is improved to include support for ServiceMonitor creation. As a result, you can now gather metrics from Rollouts using the enhanced RolloutManager controller, and these metrics are visible in monitoring. link:https://issues.redhat.com/browse/GITOPS-3271[GITOPS-3271]

Reviewer 5: ```suggestion
* Before this update, the RolloutManager controller did not create ServiceMonitor. With this update, the RolloutManager controller is improved to include support for ServiceMonitor creation. As a result, you can gather metrics from Rollouts using the enhanced RolloutManager controller, and these metrics are visible in monitoring. link:https://issues.redhat.com/browse/GITOPS-3271[GITOPS-3271]
```

Author: Fixed.

[id="deprecated-features-1-13-0_{context}"]
== Deprecated and removed features

> * In {gitops-title} 1.13, {gitops-title} Application Manager command-line interface (CLI), `kam`, is deprecated and is planned to be removed in the future release. Red{nbsp}Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to {gitops-title} Application Manager CLI, `kam`, you can use Argo CD CLI that is available from {gitops-title} Operator v1.12. link:https://issues.redhat.com/browse/GITOPS-4466[GITOPS-4466]

Reviewer 5: ```suggestion
* In {gitops-title} 1.13, the {gitops-title} Application Manager CLI, `kam`, is deprecated and is planned to be removed in the future release. Red{nbsp}Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to {gitops-title} Application Manager CLI, `kam`, you can use Argo CD CLI that is available from {gitops-title} Operator v1.12. link:https://issues.redhat.com/browse/GITOPS-4466[GITOPS-4466]
```

Author: Fixed.

> * Breaking change: Before this update, Argo Rollouts instances in {gitops-title} were installed as namespace-scoped. This means that when you created a `RolloutManager` CR within a namespace, that Argo Rollouts instance would only have permissions to deploy within that specific namespace. This issue has been fixed in this update. Now, Argo Rollouts instances in {gitops-title} are installed as cluster-scoped, which means that a RolloutManager CR created within a namespace can be deployed across the entire cluster. link:https://issues.redhat.com/browse/RHDEVDOCS-6022[RHDEVDOCS-6022]

Reviewer 5: ```suggestion
* Breaking change: Before this update, Argo Rollouts instances in {gitops-title} were installed as namespace-scoped. Previously, when you created a `RolloutManager` CR within a namespace, that Argo Rollouts instance would only have permissions to deploy within that specific namespace. This issue has been fixed in this update. Now, because Argo Rollouts instances in {gitops-title} are installed as cluster-scoped, a RolloutManager CR created within a namespace can be deployed across the entire cluster. link:https://issues.redhat.com/browse/RHDEVDOCS-6022[RHDEVDOCS-6022]
```

Author: Fixed.

[source,YAML]
----
apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
 name: argocd-sample
 namespace: default
spec:
 monitoring:
   disableMetrics: true
----
+
Use the following command to scrape metrics in the Argo CD CLI:
+
[source,terminal]
----
$ oc patch argocd argocd-sample -n default --type='json' -p='[{"op": "replace", "path": "/spec/monitoring/disableMetrics", "value": true}]'
----

* With this update, the Argo Rollouts functionality in {gitops-title} is promoted from Technology Preview (TP) to General Availability (GA). link:https://issues.redhat.com/browse/GITOPS-3848[GITOPS-3848]
+
Use this feature to perform the following actions:
+
** Run cluster-scoped Argo Rollouts instead of namespace-scoped Argo Rollouts
** Route traffic by using OpenShift Routes for canary application deployment 
** Configure the Argo Rollouts CLI

* With this update, you can perform canary-based Argo Rollouts deployments by using OpenShift Routes-based traffic management. Argo Rollouts supports traffic splitting by modifying the percentage of traffic that reaches the canary or stable application version during a deployment. link:https://issues.redhat.com/browse/GITOPS-2400[GITOPS-2400]

> * With this update, the Argo CD Applications in any namespace feature is promoted from Technology Preview (TP) to General Availability (GA). link:https://issues.redhat.com/browse/GITOPS-3796[GITOPS-3796]

Author: Fixed.

+

> * With this update, the following new commands are introduced for the Apps in any namespace feature:

Author: Fixed.

> ** `argocd proj add-source-namespace`: Use this command to add source namespaces to `AppProject.Spec.SourceNamespaces` for whitelisting specific namespaces for application creation.

Reviewer 7: ```suggestion
** `argocd proj add-source-namespace`: Use this command to add source namespaces to the `.spec.sourceNamespaces` field of the `AppProject` CR to allow specific namespaces for application creation.
```
@Dhruv-Soni11 Use conscious language while writing. RH SSG recommends when possible, rewrite documentation to avoid these terms. When it is not possible to remove the terms blacklist and whitelist, replace them with one of the following alternatives: https://redhat-documentation.github.io/supplementary-style-guide/#_blacklist_and_whitelist

Author: Fixed.

> ** `argocd proj delete-source-namespace`: Use this command to remove source namespaces from `AppProject.Spec.SourceNamespaces` and blacklist specific namespaces. link:https://issues.redhat.com/browse/GITOPS-4045[GITOPS-4045]

Reviewer 7: ```suggestion
** `argocd proj delete-source-namespace`: Use this command to remove source namespaces from the `.spec.sourceNamespaces` field of the `AppProject` CR to exclude specific namespaces. link:https://issues.redhat.com/browse/GITOPS-4045[GITOPS-4045]
```
@Dhruv-Soni11 Use conscious language while writing. RH SSG recommends when possible, rewrite documentation to avoid these terms. When it is not possible to remove the terms blacklist and whitelist, replace them with one of the following alternatives: https://redhat-documentation.github.io/supplementary-style-guide/#_blacklist_and_whitelist

Author: Fixed.

* With this update, you can customize permissions for a cluster-scoped Argo CD instance by creating user-defined `ClusterRoles` and `ClusterRoleBindings` for the Argo CD Application Controller and server components. link:https://issues.redhat.com/browse/GITOPS-2614[GITOPS-2614]

* With this update, you can customize the Keycloak route name in the {gitops-title} Operator. This update also enables support for complex routing scenarios, such as dividing routes among various ingress controller shards. link:https://issues.redhat.com/browse/GITOPS-4451[GITOPS-4451]

> * With this update, the default TLS termination policy of the Argo CD server's Route changed from `Passthrough` to `Reencrypt`. As a result, the TLS connection relies on the default ingress Operator certificate instead of the self-signed certificate generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]

Reviewer 7: ```suggestion
* With this update, the default TLS termination policy of the Argo CD server's route changed from `Passthrough` to `Reencrypt`. As a result, the TLS connection relies on the default ingress Operator certificate instead of the self-signed certificate generated by the {gitops-title} Operator. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]
```
@Dhruv-Soni11 alternatively, see the Note section in the Setting up and Argo CD instance assembly: https://77298--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_instance/setting-up-argocd-instance.html

If you prefer what is mentioned there in the Note section to what is mentioned in the RN Text here, feel free to replace it. Thanks!

Author: Fixed.

* Before this update, users were unable to connect to remote Git repositories through a SOCKS5 proxy server because it was only compatible with HTTPS URLs. With this update, users can access remote Git repositories by using a SOCKS5 proxy server that supports the SSH protocol. link:https://issues.redhat.com/browse/GITOPS-3710[GITOPS-3710]

* Before this update, the `forceHttpBasicAuth` repository configuration was not initialized properly during repository updates from Argo CD UI. As a result, the value was reset to default, leading to connection errors, particularly for Azure DevOps repositories. This update fixes the issue by adding a change in the upstream to ensure that the `forceHttpBasicAuth` configuration is correctly initialized during repository update operations. link:https://issues.redhat.com/browse/GITOPS-3706[GITOPS-3706]

* Before this update, the Argo CD CLI would designate the initial source as the main source for multi-source applications. With this update, the Argo CD CLI supports multi-source applications by considering all sources during processing. link:https://issues.redhat.com/browse/GITOPS-2623[GITOPS-2623]

> * Before this update, the RolloutManager controller did not create ServiceMonitor. With this update, the RolloutManager controller is improved to include support for ServiceMonitor creation. As a result, you can gather metrics from Rollouts using the enhanced RolloutManager controller, and these metrics are visible in monitoring. link:https://issues.redhat.com/browse/GITOPS-3271[GITOPS-3271]

Reviewer 7: @Dhruv-Soni11 It is either the Argo Rollouts controller CR or RolloutManager custom resource (CR). CHeck what exactly to use in this case. There is no such thing to use as "RolloutManager controller".

Also please check on ServiceMonitor. Must be mentioned in within backticks. Also, identify what API object it is and mention it here. Refer to https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting.

Author: Fixed as Argo Rollouts controller CR.

[id="deprecated-features-1-13-0_{context}"]
== Deprecated and removed features

> * In {gitops-title} 1.13, the {gitops-title} Application Manager CLI, `kam`, is deprecated and is planned to be removed in the future release. Red{nbsp}Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to {gitops-title} Application Manager CLI, `kam`, you can use Argo CD CLI that is available from {gitops-title} Operator v1.12. link:https://issues.redhat.com/browse/GITOPS-4466[GITOPS-4466]

Reviewer 7: ```suggestion
* In {gitops-title} 1.13, the {gitops-title} Application Manager CLI, `kam`, is deprecated and is planned to be removed in the future release. Red{nbsp}Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to the {gitops-title} Application Manager CLI, `kam`, you can use the Argo CD CLI that is available from {gitops-title} Operator v1.12. link:https://issues.redhat.com/browse/GITOPS-4466[GITOPS-4466]
```

Author: Fixed.

> * Breaking change: Before this update, Argo Rollouts instances in {gitops-title} were installed as namespace-scoped. Previously, when you created a `RolloutManager` CR within a namespace, that Argo Rollouts instance would only have permissions to deploy within that specific namespace. This issue has been fixed in this update. Now, because Argo Rollouts instances in {gitops-title} are installed as cluster-scoped, a RolloutManager CR created within a namespace can be deployed across the entire cluster. link:https://issues.redhat.com/browse/RHDEVDOCS-6022[RHDEVDOCS-6022]

Reviewer 7: @Dhruv-Soni11 Breaking change must go like this: https://docs.openshift.com/container-platform/4.7/cicd/gitops/gitops-release-notes.html#breaking-change-1-1_gitops-release-notes

Author: Fixed.

+
* With this update, the following new commands are introduced for the Argo CD applications in non-control plane namespaces feature:
** `argocd proj add-source-namespace`: Use this command to add source namespaces to the `.spec.sourceNamespaces` field of the `AppProject` CR to allow specific namespaces for application creation.
** `argocd proj delete-source-namespace`: Use this command to remove source namespaces from the `.spec.sourceNamespaces` field of the `AppProject` CR to exclude specific namespaces. link:https://issues.redhat.com/browse/GITOPS-4045[GITOPS-4045]

* With this update, you can customize permissions for a cluster-scoped Argo CD instance by creating user-defined `ClusterRoles` and `ClusterRoleBindings` for the Argo CD Application Controller and server components. link:https://issues.redhat.com/browse/GITOPS-2614[GITOPS-2614]

* With this update, you can customize the Keycloak route name in the {gitops-title} Operator. This update also enables support for complex routing scenarios, such as dividing routes among various ingress controller shards. link:https://issues.redhat.com/browse/GITOPS-4451[GITOPS-4451]

> * With this update, the route TLS termination is set as default to the reencrypt mode for both the default and user-defined Argo CD instances. TLS connections to the Argo CD instances now receive the default ingress certificate that is set in OpenShift Container Platform, instead of the self-signed Argo CD certificate generated by the {gitops-title} Operator. You can modify the route TLS termination policy by configuring the ``.spec.server.route.tls` field of the Argo CD CR. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]

Reviewer 7: ```suggestion
* With this update, the route TLS termination is set as default to the reencrypt mode for both the default and user-defined Argo CD instances. TLS connections to the Argo CD instances now receive the default ingress certificate that is set in OpenShift Container Platform, instead of the self-signed Argo CD certificate generated by the {gitops-title} Operator. You can modify the route TLS termination policy by configuring the `.spec.server.route.tls` field of the Argo CD CR. link:https://issues.redhat.com/browse/GITOPS-3918[GITOPS-3918]
```

Gives a rendering issue also: 
![image](https://github.com/openshift/openshift-docs/assets/77115114/f7185e87-1354-4ec2-9bb9-be81f93f5ff8)

Author: Fixed.

* Before this update, the Argo Rollouts controller CR did not create `ServiceMonitor`. With this update, the Argo Rollouts controller CR is improved to include support for `ServiceMonitor` creation. As a result, you can gather metrics from Rollouts using the enhanced Argo Rollouts controller CR, and these metrics are visible in monitoring. link:https://issues.redhat.com/browse/GITOPS-3271[GITOPS-3271]

[id="deprecated-features-1-13-0_{context}"]
== Deprecated and removed features

* In {gitops-title} 1.13, the {gitops-title} Application Manager CLI, `kam`, is deprecated and is planned to be removed in the future release. Red{nbsp}Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to the {gitops-title} Application Manager CLI, `kam`, you can use the Argo CD CLI that is available from {gitops-title} Operator v1.12. link:https://issues.redhat.com/browse/GITOPS-4466[GITOPS-4466]

[id="breaking-change-1-13-0_{context}"]
== Breaking change

> === Argo Rollouts instances are installed as namespace-scoped

Reviewer 7: @Dhruv-Soni11 For any Level 1 or lower subsections in an assembly (for example, ==, ===, etc.), add a _{context} variable to the end of anchor IDs: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-subsections-level-1-and-lower-headings

Author: Fixed.

[id="deprecated-features-1-13-0_{context}"]
== Deprecated and removed features

* In {gitops-title} 1.13, the {gitops-title} Application Manager CLI, `kam`, is deprecated and is planned to be removed in the future release. Red{nbsp}Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. As an alternative to the {gitops-title} Application Manager CLI, `kam`, you can use the Argo CD CLI that is available from {gitops-title} Operator v1.12. link:https://issues.redhat.com/browse/GITOPS-4466[GITOPS-4466]

[id="breaking-change-1-13-0_{context}"]
== Breaking change

[id="argo-rollouts-instances-are-installed-as-namespace-scoped_{context}"]
=== Argo Rollouts instances are installed as namespace-scoped

> * When you upgrade to {gitops-title} v1.13, Argo Rollouts instances in {gitops-title} are installed as namespace-scoped. Previously, when you created a `RolloutManager` CR within a namespace, that Argo Rollouts instance would only have permissions to deploy within that specific namespace. Now, because Argo Rollouts instances in {gitops-title} are installed as cluster-scoped, a RolloutManager CR created within a namespace can be deployed across the entire cluster. link:https://issues.redhat.com/browse/RHDEVDOCS-6022[RHDEVDOCS-6022]

Reviewer 7: ```suggestion
* When you upgrade to {gitops-title} v1.13, Argo Rollouts instances in {gitops-title} are installed as namespace-scoped. Previously, when you created a `RolloutManager` CR within a namespace, that Argo Rollouts instance would only have permissions to deploy within that specific namespace. Now, because Argo Rollouts instances in {gitops-title} are installed as cluster-scoped, a RolloutManager CR created within a namespace can be deployed across the entire cluster. link:https://issues.redhat.com/browse/GITOPS-3847[GITOPS-3847]
```

Author: Fixed.
